crates/sui-bridge/src/action_executor.rs

```diff
@@ -0,0 +1,894 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! BridgeActionExecutor receives BridgeActions (from BridgeOrchestrator),
+//! collects bridge authority signatures and submit signatures on chain.
+
+use mysten_metrics::spawn_logged_monitored_task;
+use shared_crypto::intent::{Intent, IntentMessage};
+use sui_json_rpc_types::{
+    SuiExecutionStatus, SuiTransactionBlockEffects, SuiTransactionBlockEffectsAPI,
+};
+use sui_types::{
+    base_types::{ObjectID, ObjectRef, SuiAddress},
+    committee::VALIDITY_THRESHOLD,
+    crypto::{Signature, SuiKeyPair},
+    digests::TransactionDigest,
+    object::Owner,
+    programmable_transaction_builder::ProgrammableTransactionBuilder,
+    transaction::{Transaction, TransactionData},
+};
+
+use crate::{
+    client::bridge_authority_aggregator::BridgeAuthorityAggregator,
+    error::BridgeError,
+    storage::BridgeOrchestratorTables,
+    sui_client::{SuiClient, SuiClientInner},
+    types::{BridgeAction, VerifiedCertifiedBridgeAction},
+};
+use std::sync::Arc;
+use tracing::{error, info, warn};
+
+pub const CHANNEL_SIZE: usize = 1000;
+
+#[derive(Debug)]
+pub struct BridgeActionExecutionWrapper(pub BridgeAction, pub u64);
+
+pub trait BridgeActionExecutorTrait {
+    fn run(
+        self,
+    ) -> (
+        Vec<tokio::task::JoinHandle<()>>,
+        mysten_metrics::metered_channel::Sender<BridgeActionExecutionWrapper>,
+    );
+}
+
+pub struct BridgeActionExecutor<C> {
+    sui_client: Arc<SuiClient<C>>,
+    bridge_auth_agg: Arc<BridgeAuthorityAggregator>,
+    key: Option<SuiKeyPair>,
```

Comment:
Why Option?

Reply:
it's from past iterations, not longer needed, will remove the option